File: dynabuffers-java/src/test/kotlin/dynabuffers/usecase/Schema07Test.kt
package dynabuffers.usecase

import dynabuffers.AbstractDynabuffersTest
import dynabuffers.Dynabuffers
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test

class Schema07Test : AbstractDynabuffersTest() {

    @Test
    fun testParse() {
        val imageBytes = Schema07Test::class.java.getResourceAsStream("/_data/1.jpg").readAllBytes()
        val engine = Dynabuffers.parse(Schema07Test::class.java.getResourceAsStream("/schema07.dbs"))
        val output = mapOf("image" to imageBytes)
        val result = engine.serialize(output)
        val serialized = engine.deserialize(result)

        Assertions.assertThat(result.size < 25_000).isTrue()
        Assertions.assertThat(serialized["image"]).isEqualTo(imageBytes)
    }

}

File: dynabuffers-python/tests/usecase/Schema07Test.py
import os
import unittest

from antlr4 import FileStream

from dynabuffers.Dynabuffers import Dynabuffers


class Schema05Test(unittest.TestCase):
    root_dir = os.path.dirname(os.path.realpath(__file__))

    def test_parse(self):
        with open(self.root_dir + "/1.jpg", 'rb') as f:
            data = b"".join(f.readlines())
            engine = Dynabuffers.parse(FileStream(self.root_dir + "/schema07.dbs"))
            map = engine.deserialize(engine.serialize({"image": data}))

            self.assertEqual(map, {"image": data})


if __name__ == "__main__":
    unittest.main()
